### ISSUE TITLE
why TabularPrediction can not utilize GPU?

### ISSUE THREAD

**Issue by Gpwner** [awslabs/autogluon#894]
Here is my code,I just dont understand why it can  not utilize gpu  at all:
```
import autogluon.core as ag
from autogluon.tabular import TabularPrediction as task
dir = 'regression_model'  # specifies folder where to store trained models
predictor = task.fit(train_data=train_embedding_DF, label='FAILED_RATIO', output_directory=dir,ngpus_per_trial=1)
```
And my CPU utilization is upon to 99%.

**Comment by Gpwner** [comment]
And I got this error:
```
Fitting model: NeuralNetMXNet ...
	Warning: Exception caused NeuralNetMXNet to fail during training... Skipping this model.
Traceback (most recent call last):
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/autogluon/tabular/trainer/abstract_trainer.py", line 866, in _train_and_save
    model = self._train_single(X_train, y_train, model, X_val, y_val, **model_fit_kwargs)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/autogluon/tabular/trainer/abstract_trainer.py", line 838, in _train_single
    model.fit(X_train=X_train, y_train=y_train, X_val=X_val, y_val=y_val, **model_fit_kwargs)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/autogluon/tabular/models/abstract/abstract_model.py", line 291, in fit
    self._fit(**kwargs)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/autogluon/tabular/models/tabular_nn/tabular_nn_model.py", line 204, in _fit
    self.train_net(train_dataset=train_dataset, params=params, val_dataset=val_dataset, initialize=True, setup_trainer=True, time_limit=time_limit, reporter=reporter)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/autogluon/tabular/models/tabular_nn/tabular_nn_model.py", line 327, in train_net
    output = self.model(data_batch)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/mxnet/gluon/block.py", line 682, in __call__
    out = self.forward(*args)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/autogluon/tabular/models/tabular_nn/embednet.py", line 226, in forward
    return nd.sigmoid(unscaled_pred) * self.y_span + self.y_lower
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/mxnet/ndarray/ndarray.py", line 314, in __mul__
    return multiply(self, other)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/mxnet/ndarray/ndarray.py", line 3758, in multiply
    None)
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/mxnet/ndarray/ndarray.py", line 3571, in _ufunc_helper
    return fn_array(lhs, rhs)
  File "<string>", line 52, in broadcast_mul
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/mxnet/_ctypes/ndarray.py", line 91, in _imperative_invoke
    ctypes.byref(out_stypes)))
  File "/home/user/anaconda3/envs/autogluon/lib/python3.7/site-packages/mxnet/base.py", line 246, in check_call
    raise get_last_ffi_error()
mxnet.base.MXNetError: Traceback (most recent call last):
  File "src/imperative/./imperative_utils.h", line 72
MXNetError: Check failed: inputs[i]->ctx().dev_mask() == ctx.dev_mask() (1 vs. 2) : Operator broadcast_mul require all inputs live on the same context. But the first argument is on gpu(0) while the 2-th argument is on cpu(0)
Traceback (most recent call last):
  File "src/imperative/./imperative_utils.h", line 72
MXNetError: Check failed: inputs[i]->ctx().dev_mask() == ctx.dev_mask() (1 vs. 2) : Operator broadcast_mul require all inputs live on the same context. But the first argument is on gpu(0) while the 2-th argument is on cpu(0)

```

**Comment by Innixma** [comment]
In the upcoming v0.1 release, GPU will no longer need to be specified in order to be utilized when appropriate. Additionally, numerous stability improvements related to GPU have been added for v0.1, so hopefully this issue will be fixed in v0.1. v0.1 is planned to release in ~2 weeks, or you can take a look at and install the pre-release now: https://auto.gluon.ai/dev/index.html

**Comment by Gpwner** [comment]
> In the upcoming v0.1 release, GPU will no longer need to be specified in order to be utilized when appropriate. Additionally, numerous stability improvements related to GPU have been added for v0.1, so hopefully this issue will be fixed in v0.1.
> 
> One thing to note is that for tabular datasets, GPU often is not much faster than CPU (or slower) due to the smaller scale networks that are used. I would recommend trying with only CPU and seeing if the models train at acceptable speeds for your use-case.
> 
> v0.1 is planned to release in ~2 weeks, or you can take a look at and install the pre-release now: https://auto.gluon.ai/dev/index.html

Thanks for reply.

**Comment by Innixma** [comment]
Hi @Gpwner , I have added support for GPU in #896, which will be available in pre-release and in v0.1.

To ensure GPU is used, add the argument `ag_args={'num_gpus': 1}` to the call to `TabularPredictor.fit()`
